File: cpp/CyclicRotation/cyclic_rotation.cpp
#include "cyclic_rotation.h"

std::vector<int> cyclic_rotation::solution(std::vector<int>& A, int K)
{
    if (A.empty() || A.size() == 1)
        return A;

    K = K % A.size();

    if (K == 0)
        return A;

    std::rotate(A.rbegin(), A.rbegin() + K, A.rend());
    return A;
}

File: csharp/CyclicRotation/CyclicRotation.cs
public static class CyclicRotation
{
    public static int[] Solution(int[] a, int k)
    {
        if (a.Length <= 0 || k <= 0)
            return a;

        var b = new int[a.Length];
        b[0] = a[^1];

        for (int i = 0; i < a.Length - 1; i++)
            b[i + 1] = a[i];

        return k > 1 ? Solution(b, k - 1) : b;
    }
}